### PR TITLE
Fix sporadic errors in test_results due to tutorial popup

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://progress.opensuse.org/issues/67660
First Verification: https://openqa.opensuse.org/t1287908

### CODE REVIEW

**Comment by okurz** [comment]
If you want to fix sporadic results I recommend you test more than one run – and the one you referenced failed anyway.

Could we please try to find a fix that is not going back to the old way of checking screen multiple times and relying on timing behaviour?

**Comment by okurz** [comment]
https://openqa.opensuse.org/tests/1285301#step/test_results/2 shows that "All Tests" is clicked but in the same screen "dont-notify-me" is detected. Either we add a wait_still_screen there or – probably better – include the "reload" button from the browser bar or the favicon area in the needle to not match while a screen is loading.

**Comment by asdil12** [comment]
Detecting the browser loading indicatior doesn't sound very clean to me and could easily break.
I would rather waste the worker's time than the time of a human reviewer.

**Comment by okurz** [comment]
> I would rather waste the worker's time than the time of a human reviewer.

No doubt about that. I am not sure your approach will be more stable though. But it's ok this way and we can try

LGTM

**Comment by asdil12** [comment]
More verification:
https://openqa.opensuse.org/tests/overview?version=Tumbleweed&distri=opensuse&build=asdil12%2Fos-autoinst-distri-opensuse%2310435

**Comment by asdil12** [comment]
It looks much better now (I also fixed some needle problem).

File: tests/openqa/webui/test_results.pm [hunk 1]
```diff
@@ -38,6 +38,12 @@ sub upload_autoinst_log {
 
 sub handle_notify_popup {
     return undef unless match_has_tag 'openqa-dont-notify-me';
+    for my $i (1 .. 5) {
+        assert_and_click 'openqa-dont-notify-me';
+        if (check_screen('openqa-tutorial-confirm')) {
```
Comment (foursixnine) [inline tests/openqa/webui/test_results.pm:43]: check screen also for `openqa-dont-notify-toggled`, and if so, click the confirm? https://openqa.opensuse.org/tests/1288011#step/test_results/7

Comment (asdil12) [reply tests/openqa/webui/test_results.pm:43]: The confirm button is only visible when the checkbox is already toggled.

Comment (foursixnine) [reply tests/openqa/webui/test_results.pm:43]: Yeah, but in the test that I reference, the button is shown already :P, and openQA is looking for the unchecked box...

Comment (asdil12) [reply tests/openqa/webui/test_results.pm:43]: Ah - seems that `check_screen` doesn't have any timeout which leads to thes race condition.
Adding a timeout should do it.

**Comment by asdil12** [comment]
> I was fine with the `check_screen` without timeout but now again you add back a rather inefficient approach wasting lot of time waiting for nothing to happen. Please use proper needles which exclude each other and multi-tag assert_screen calls

That might still lead to a race condition if the screen is not updated fast enough (which is exactly the reason of the current race condition).
Waiting a bit is exactly what needs to be done here and I honestly don't want to discuss more than 30 seconds about 30 seconds of waiting on a worker.

**Comment by okurz** [comment]
It's not just 30s if you consider TIMEOUT_SCALE and a for loop doing up to 5x check_screen. And there can be cases where even a waiting time of 30s is not enough, if the server is a bit slow to respond so to put it to extreme you would need to extend the waiting time even more which is not what I suggest to do. I would be ok if you put in a `wait_still_screen` call which I have proposed. Your approach shows bad practice in general software development: When not knowing what exactly to wait for you just wait (and waste) an arbitrary time which sums up and is also seen in other parts of this repo. If you want to invest your time discussing it or not is your choice. I already gave specific suggestions in code and regarding needle choices what to do.

**Comment by asdil12** [comment]
> It's not just 30s if you consider TIMEOUT_SCALE and a for loop doing up to 5x check_screen. And there can be cases where even a waiting time of 30s is not enough, if the server is a bit slow to respond so to put it to extreme you would need to extend the waiting time even more which is not what I suggest to do. I would be ok if you put in a `wait_still_screen` call which I have proposed. Your approach shows bad practice in general software development: When not knowing what exactly to wait for you just wait (and waste) an arbitrary time which sums up and is also seen in other parts of this repo. If you want to invest your time discussing it or not is your choice. I already gave specific suggestions in code and regarding needle choices what to do.

I don't see any advantage in `wait_still_screen` as it would just *waste time* as well and I would still need to check for the needle afterwards.
As only ~5% of my verification runs even ran into this race condition, only in these cases there would even be a larger abount of the available timeout actually consumed.

**Comment by asdil12** [comment]
I wonder, how https://openqa.opensuse.org/tests/1289107#step/test_results/18 can happen.
The needle has an exclude area above the differing spot but is still not getting 100%.

**Comment by okurz** [comment]
hm, I also don't see any obvious difference in https://openqa.opensuse.org/tests/1289107#step/test_results/18 . Maybe there is very subtle change in shading. You might be able to avoid the exclude area with just covering the test module name and the green color of "passed" without the text

**Comment by asdil12** [comment]
https://openqa.opensuse.org/t1302713

**Comment by asdil12** [comment]
https://openqa.opensuse.org/t1306024

**Comment by asdil12** [comment]
https://openqa.opensuse.org/t1307293

**Pull request merged**